arch/common/gen_isr_tables.py

```diff
@@ -142,10 +181,10 @@ def write_source_file(fp, vt, swt, intlist, syms):
     nv = intlist["num_vectors"]
 
     if vt:
-        fp.write("uintptr_t __irq_vector_table _irq_vector_table[%d] = {\n" % nv)
-        for i in range(nv):
-            fp.write("\t{},\n".format(vt[i]))
-        fp.write("};\n")
+        if "CONFIG_IRQ_VECTOR_TABLE_JUMP_BY_ADDRESS" in syms:
+            write_c_irq_vector_table(fp, vt, nv)
+        else:
+            write_assembly_irq_vector_table(fp, vt, nv, syms)
```

Comment:
it would be better to explicitly check for `CONFIG_IRQ_VECTOR_TABLE_JUMP_BY_ADDRESS` and `CONFIG_IRQ_VECTOR_TABLE_JUMP_BY_CODE` and assert otherwise.

Reply:
done